File: frontend/src/types/exam.ts
import type { Link } from './hateoas';

export interface Exam {
  id: string;
  patientId: string;
  type: string;
  scheduledDate: string;
  status: string;
  results: string | null;
  notes: string | null;
  links: Link[];
}

export interface ExamList {
  items: Exam[];
  links: Link[];
}

export interface CreateExamRequest {
  patientId: string;
  type: string;
  scheduledDate: string;
  status: string;
  results: string | null;
  notes: string | null;
}

export interface UpdateExamRequest {
  type: string;
  scheduledDate: string;
  status: string;
  results: string | null;
  notes: string | null;
}

File: RestReactAspire.Server/Models/Exam.cs
namespace RestReactAspire.Server.Models;

public class Exam
{
    public Guid Id { get; set; }
    public Guid PatientId { get; set; }
    public Guid? DoctorId { get; set; }
    public required string Type { get; set; }
    public DateOnly ScheduledDate { get; set; }
    public required string Status { get; set; }
    public string? Results { get; set; }
    public string? Notes { get; set; }
}
